.github/workflows/ci.yml

```
@@ -0,0 +1,39 @@
+name: CI
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-24.04
+    timeout-minutes: 10
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+      - uses: cachix/install-nix-action@v27
+        with:
+          install_url: https://releases.nixos.org/nix/nix-2.23.3/install
+          github_access_token: "${{ secrets.CI_BUILD_TOKEN }}"
```

Let's update the action version and credentials settings
```suggestion
      # v37.1.0
      - uses: cachix/install-nix-action@d1ca217b388ee87b2507a9a93bf01368bde7cec2
        with:
          install_url: https://releases.nixos.org/nix/nix-2.23.3/install
          github_access_token: "${{ secrets.GITHUB_TOKEN }}"
```